docs/content/config/security/mail_crypt.md

```diff
@@ -64,3 +70,71 @@ Official Dovecot documentation: https://doc.dovecot.org/configuration_manual/mai
     ```
 
 This should be the minimum required for encryption of the mail while in storage.
+
+
+## Encrypted User Keys
+
+1. Create `10-custom.conf` and populate it with the following:
+
+    ```
+    mail_attribute_dict = file:%h/Maildir/dovecot-attributes
+    mail_plugins = $mail_plugins mail_crypt
+    mail_debug= yes
+
+    plugin {
+      mail_crypt_curve = secp521r1
+      mail_crypt_save_version = 2
+      mail_crypt_require_encrypted_user_key = yes
+    }
+    ```
+
+2. Create `auth-passwdfile.inc` and populate it with the following:
```

Comment:
I saw that the template language of dovecot has conditionals, could we maybe detect that the mail_crypt plugin is loaded (or some ENV variable that is set) and just change the default `auth-passwdfile.inc` in the container? Otherwise if the repo changes that file, stuff will break for everyone that took a copy of it (like described here in the manual?)

Reply:
Sorry, yes, those weren't supposed to be added to the repo. They were just for my testing.

Reply:
ah true, but I was reading the documentation, there it also says: setup this custom file. the same point remains, or have you been able to integrate it all, and is the documentation just running behind a bit? :)

Reply:
Ah yes, docs were behind.
Still trying to figure out how some tests are failing.

Reply:
Sorry to start about it again, my point is. This doc says: override this file. Since this file is also in the main docker container: https://github.com/docker-mailserver/docker-mailserver/blob/master/target/dovecot/auth-passwdfile.inc

This is not an ideal solution, as it will lead to co-evolution changes to this file. If we can find a way to change the build-in `auth-passwdfile.inc` to detect the `mail-crypt-plugin`? Than this manual gets a lot easier, and the containers easier to maintain

Reply:
Thanks for clarifying. Give me a few days to consider how to do this and get back. I think the issue with that is now the user encryption keys/password stuff becomes an official feature -- which I'm totally fine with but maybe the maintainers don't agree! This would make sense since I added features in setup.sh for it, so it's already sort of a built-in feature in many ways. We can easily allow users to set an ENV to enable it and then even change the setup.sh to check for that ENV and automatically run the doveadm commands instead of requiring users to set `-g` or `-c`.

Reply:
Hey @polarathene , when you start your review, this is probably the place to start. NO RUSH!! :)

Reply:
Hey @DavyLandman, I've reworked it to be a built-in feature you can enable with an ENV. 

However, I think there is still a way to simplify this into a single .inc file. I couldn't find any way to do conditionals in .inc files though and I wonder if you have any links or advice for it. I'm usually pretty good at finding this stuff using google 😅

Reply:
Currently on vacation but it sounds good. I remember there are some conditional expression inside dovecot config, and I thought that maybe those could be used to switch between 0 and 2 for encryption type?

Reply:
Like this feature: https://doc.dovecot.org/configuration_manual/config_file/config_variables/#conditionals

Reply:
Just since this is a big PR, adding specific ping  to @NorseGaud for this.

(Keep up the good work btw!)

Reply:
I'm going to resolve this for now. I'm unsure exactly what is necessary to achieve a conditional approach. A separate PR can always be submitted to expand the feature.

Reply:
True, good work on finishing this PR!